app/javascript/vue/books-data.js

```diff
@@ -1,7 +1,27 @@
 import axios from 'axios'
+import _ from 'lodash'
 
-export default new Promise((resolve, reject) => {
-  axios('/api/books')
-    .then((response) => resolve(response.data))
-    .catch((e) => reject(e))
-})
+function getAll () {
+  return new Promise((resolve, reject) => {
+    axios('/api/books')
+      .then((response) => resolve(response.data))
+      .catch((e) => reject(e))
+  })
+}
+
+function getById (id) {
+  return new Promise((resolve, reject) => {
+    getAll()
+      .then((books) => {
+        const book = _.find(books, { id: id })
+        if (book) {
+          resolve(book)
+        } else {
+          reject(new Error(`'Can't find book with ID ${id}'`))
```

Comment:
@isabloCh you don't need `'` quotes since you already use  ` quote

Reply:
are needed
![image](https://user-images.githubusercontent.com/28955056/46872145-0f02c700-ce3c-11e8-84ba-e1f9f7732a8f.png)

Reply:
@isabloCh the quotes should be like these: `...` since you inject data. In your example you use '...'

Reply:
or display the message without quotes

Reply:
+